pkg/controllers/management/auth/globalrolebinding_handler.go

```diff
@@ -51,6 +70,8 @@ func (grb *globalRoleBindingLifecycle) Remove(obj *v3.GlobalRoleBinding) (runtim
 }
 
 func (grb *globalRoleBindingLifecycle) reconcileGlobalRoleBinding(globalRoleBinding *v3.GlobalRoleBinding) error {
+	var templateRuleExists, templateVersionRuleExists bool
```

Comment:
I don't think there is a need for bool and objects. We can just use pointer type for catalogTemplateRule and catalogTemplateVersionRule if you need to figure out the presence of either. Then you can use & when passing down the value